apps/src/templates/sectionProgress/StudentProgressDetailCell.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { PropTypes, Component } from 'react';
+import ProgressBubbleSet from '@cdo/apps/templates/progress/ProgressBubbleSet';
+import { levelByLesson } from '@cdo/apps/code-studio/progressRedux';
+
+const styles = {
+  bubbles: {
+    overflowX: 'scroll',
+    whiteSpace: 'nowrap',
+  }
+};
+
+export default class StudentProgressDetailCell extends Component {
+  static propTypes = {
+    section: PropTypes.shape({
+      id: PropTypes.number.isRequired,
+      students: PropTypes.arrayOf(PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        name: PropTypes.string.isRequired,
+      })).isRequired
+    }).isRequired,
+    studentId: PropTypes.number.isRequired,
+    stageId: PropTypes.number.isRequired,
+    scriptData: PropTypes.shape({
+      stages: PropTypes.arrayOf(PropTypes.shape({
+        levels: PropTypes.arrayOf(PropTypes.object).isRequired
+      })),
+      id: PropTypes.number.isRequired,
+    }).isRequired,
+    studentLevelProgress: PropTypes.objectOf(
+      PropTypes.objectOf(PropTypes.number)
+    ).isRequired,
+  };
+
+  studentLevelProgressInStage(studentId, stageId) {
+    const { scriptData, studentLevelProgress } = this.props;
+
+    const fakeState = {
```

Comment:
better variable name?

Reply:
I don't totally understand what this is or why it exists so I don't know how to name it.  Do you have a clear sense of why it was done this way in the first place?

Reply:
Since we're using `ProgressBubbleSet` and `levelByLesson`, which wasn't built specifically for what we're using, it has a different format of props that it expects, including state, levelProgress, and currentLevelId. We don't need all of those to render what we need, but we do need to pass the component the correct shape of prop, so we build this `fakeState`. Since we're passing it to levels on line 52, we could call it `levelState`